pallets/xcm-helper/src/lib.rs

```diff
@@ -0,0 +1,595 @@
+//!
+//! - [`Config`]
+//! - [`Call`]
+//! - [`Pallet`]
+//!
+//! ## Overview
+//!
+//! XCM Helper Pallet provides following functionalities:-
+//!
+//! - Handling withdrawal requests from Relayers.
+//! - Managing Thea Public Key.
+//! - Parachain asset management.
+//! - Executing Withdrawal request every block.
+//!
+//! ## Terminology
+//!
+//! - **Thea key** Thea Key is Multi-party ECDSA Public Key which has access to transfer funds from
+//!   Polkadex Sovereign Accounts to Others on Native/Foreign Chain using XCMP.
+//!
+//! - **WithdrawalExecutionBlockDiff** Delays in Blocks after which Pending withdrawal will be executed.
+//!
+//! - **TheaMessage** Thea Messages will be fetched and relayed by Relayers from Parachain to Solochain.
+//!
+//! - **ParachainAsset** Type using which native Parachain will identify assets from foregin Parachain.
+//!
+//! ### Implementations
+//! The XCM Helper pallet provides implementations for the following traits. If these traits provide
+//! the functionality that you need, then you can avoid coupling with the XCM Helper pallet.
+//!
+//! -[`TransactAsset`]: Used by XCM Executor to deposit, withdraw and transfer native/non-native asset on Native Chain.
+//! -[`AssetIdConverter`]: Converts Assets id from Multilocation Format to Local Asset Id and vice-versa.
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Functions
+//! - `withdraw_asset` - Transfers Assets from Polkadex Sovereign Account to Others on native/non-native parachains using XCMP.
+//! - `change_thea_key` - Replaces existing Thea Key with new one.
+//! - `set_thea_key` - Initializes Thea Key.
+//! - `create_parachain_asset` - Creates new Assets using Parachain info.
+//!
+//! ### Supported Origins
+//! - `AssetCreateUpdateOrigin` - Origin which has access to Create Asset.
+//!
+//! ### Public Functions
+//! - `handle_deposit` - Handles deposits from foreign chain.
+//! - `deposit_native_asset` - Deposits Native Assets using Balances Pallet.
+//! - `deposit-non-native-asset` - Deposits Non-native Assets using Assets Pallet.
+//!
+//! ### Public Inspection functions - Immutable (accessors)
+//! - `get_pallet_id` - Get xcm_helper Pallet Id
+//! - `get_destination_account` - Converts Multilocation to AccountId.
+//! - `is_polkadex_parachain_destination` - Checks if destination address belongs to native parachain or not.
+//! - `is_parachain_asset` - Checks if given asset is native asset or not.
+//!
+//! ### Storage Items
+//! - `ActiveTheaKey` - Stores Latest Thea Key.
+//! - `WithdrawNonce` - Stores Latest withdrawal nonce.
+//! - `PendingWithdrawals` - Stores all pending withdrawal.
+//! - `FailedWithdrawals` - Stores failed withdrawals which failed during execution.
+//! - `TheaAssets` - Stores all Thea Assets.
+//! -
+//! # Events
+//! - `AssetDeposited` - Asset Deposited from XCM.
+//! - `AssetWithdrawn` - Asset burned/locked from native Parachain.
+//! - `TheaAssetCreated` - New Asset Created.
```

Comment:
This documentation is outdated, multiple members from this doc were removed. Please update it.

Reply:
Update could be found in the latest commits from me.